flink-connectors/flink-connector-hive/src/main/java/org/apache/flink/table/catalog/hive/HiveCatalog.java

```diff
@@ -313,6 +327,20 @@ public void createTable(ObjectPath tablePath, CatalogBaseTable table, boolean ig
 			throw new DatabaseNotExistException(getName(), tablePath.getDatabaseName());
 		}
 
+		if (table instanceof ConnectorCatalogTable) {
+			ConnectorCatalogTable connectorTable = connectorTables.get(tablePath);
+
+			if (connectorTable != null && !tableExistsInHiveMetastore(tablePath)) {
```

Comment:
Why cannot we just call tableExists()?

Reply:
Good catch! Not sure why I did it that way.... Changed